tests/csapi/account_deactivate_test.go

```diff
@@ -16,6 +20,41 @@ func TestDeactivateAccount(t *testing.T) {
 	password := "superuser"
 	authedClient := deployment.RegisterUser(t, "hs1", "test_deactivate_user", password, false)
 	unauthedClient := deployment.Client(t, "hs1", "")
+
+	// Ensure that the first step, in which the client queries the server's user-interactive auth flows, returns
+	// at least one auth flow involving a password.
+	t.Run("Password flow is available", func(t *testing.T) {
+		reqBody := client.WithJSONBody(t, map[string]interface{}{})
+		res := authedClient.DoFunc(t, "POST", []string{"_matrix", "client", "v3", "account", "deactivate"}, reqBody)
+		must.MatchResponse(t, res, match.HTTPResponse{
+			StatusCode: 401,
+			JSON: []match.JSON{
+				match.JSONArrayEach(
```

Comment:
This feels awful. I'm pretty sure these matchers should be able to stack recursively, but I'm not sure how to — the matcher seems to want a function rather than another matcher as its argument.